ts/Dashboards/Components/Component.ts

```diff
@@ -101,57 +148,138 @@ abstract class Component<TEventObject extends Component.EventTypes = Component.E
         ],
         editableOptionsBindings: EditableOptions.defaultBindings
     };
-
+    /**
+     * The HTML element or id of HTML element that is used for appending
+     * a component.
+     */
     public parentElement: HTMLElement;
+    /**
+     * Instance of cell, where component is attached.
+     */
     public parentCell?: Cell;
+    /**
+     * Store allows you to load data via URL or from a local source.
+     */
     public store?: Component.StoreTypes; // the attached store
+    /**
+     * Size of the component (width and height).
+     */
     protected dimensions: { width: number | null; height: number | null };
+    /**
+     * The HTML element where the component is.
+     */
     public element: HTMLElement;
+    /**
+     * The HTML element where the title is.
+     */
     public titleElement?: HTMLElement;
+    /**
+     * The HTML element where the caption is.
+     */
     public captionElement?: HTMLElement;
+    /**
+     * The HTML element where the component's content is.
+     */
     public contentElement: HTMLElement;
+    /**
+     * The options for the component.
+     * */
     public options: Component.ComponentOptions;
+    /**
+     * The type of component like: `HTML`, `KPI`, `Highcharts`, `DataGrid`.
+     */
     public type: string;
+    /**
+     * Sets an ID for the component's `div`.
```

Comment:
This needs a bit more clarifiaction as a component can have multiple `div`.

Reply:
The component's id is for the main component div that is unique.

Reply:
Fixed.